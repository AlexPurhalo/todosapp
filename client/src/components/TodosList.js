import React from 'react'

export default function ({ todos, toggleTodo }) {
  return (
    <ul>
      {todos.map(todo => (
        <li onClick={() => toggleTodo(todo.id)} 
            key={todo.id} 
            style={{textDecoration: todo.isDone ? 'line-through' : ''}}
            children={todo.title} />
      ))}
    </ul>
  )
}

