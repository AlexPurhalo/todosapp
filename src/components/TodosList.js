import React from 'react'

export default function ({ todos, toggleTodo }) {
  return (
    <ul>
      {todos.map(todo => (
        <li onClick={() => toggleTodo(todo.id, todos)} 
            key={todo.id} 
            style={{textDecoration: todo.done ? 'line-through' : ''}}
            children={todo.title} />
      ))}
    </ul>
  )
}

