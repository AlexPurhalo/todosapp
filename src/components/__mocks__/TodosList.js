import React from 'react'
const TodosListMock = () => <div></div>
TodosListMock.displayName = 'TodosListMock'
export default TodosListMock
