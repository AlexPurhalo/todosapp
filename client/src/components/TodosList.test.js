import React from 'react'
import { shallow } from 'enzyme'
import TodosList from './TodosList'

describe('TodosList component', () => {
  const todos = [
    { id: 1, title: 'Call mom',       isDone: false },
    { id: 2, title: 'Clean the room', isDone: true  },
    { id: 3, title: 'Watch a movie',  isDone: true  }
  ]

  test('renders correct UI', () => {
    const wrapper = shallow(<TodosList todos={todos} />)
    expect(wrapper).toMatchSnapshot()
  })

  test('is rendering when empty todos', () => {
    const wrapper = shallow(<TodosList todos={[]}/>)
    expect(wrapper.exists()).toBeTruthy
  })

  test('contains items when todos are provided', () => {
    const wrapper = shallow(<TodosList todos={todos} />)
    expect(wrapper.find('ul').children()).toHaveLength(todos.length)
  })

  test('calls the callback responsible for completing a todo after click', () => {
    const toggleTodo = jest.fn()
    const wrapper = shallow(<TodosList todos={todos} toggleTodo={toggleTodo} />)
    wrapper.find('ul').childAt(0).simulate('click')
    expect(toggleTodo).toHaveBeenCalledWith(todos[0].id)
  })
})

