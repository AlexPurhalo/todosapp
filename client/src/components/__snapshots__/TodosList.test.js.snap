// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodosList component renders correct UI 1`] = `ShallowWrapper {}`;
